File: posixlib/src/main/resources/scala-native/linux/eventfd.c
#if defined(SCALANATIVE_COMPILE_ALWAYS) || defined(__SCALANATIVE_POSIX_EVENTFD)

#ifdef __linux__

#include <sys/eventfd.h>

int scalanative_efd_cloexec() { return EFD_CLOEXEC; }
int scalanative_efd_nonblock() { return EFD_NONBLOCK; }
int scalanative_efd_semaphore() { return EFD_SEMAPHORE; }

#endif // __linux__

#endif // __SCALANATIVE_POSIX_EVENTFD

File: posixlib/src/main/scala/scala/scalanative/posix/linux/eventfd.scala
package scala.scalanative
package linux

import unsafe._
import unsigned._

@extern
@define("__SCALANATIVE_POSIX_EVENTFD")
object eventfd {

  @name("scalanative_efd_cloexec")
  def EFD_CLOEXEC: CInt = extern
  @name("scalanative_efd_nonblock")
  def EFD_NONBLOCK: CInt = extern
  @name("scalanative_efd_semaphore")
  def EFD_SEMAPHORE: CInt = extern

  def eventfd(initval: UInt, flags: CInt): CInt = extern

}
